src/backend/contacts.nim

```diff
@@ -21,25 +21,31 @@ proc unblockContact*(id: string): RpcResponse[JsonNode] {.raises: [Exception].}
 proc removeContact*(id: string): RpcResponse[JsonNode] {.raises: [Exception].} =
   result = callPrivateRPC("removeContact".prefix, %* [id])
 
-proc rejectContactRequest*(id: string): RpcResponse[JsonNode] {.raises: [Exception].} =
-  let payload = %*[{
-    "id": id
-  }]
-  result = callPrivateRPC("rejectContactRequest".prefix, payload)
-
 proc setContactLocalNickname*(id: string, name: string): RpcResponse[JsonNode] {.raises: [Exception].} =
   let payload = %* [{
     "id": id,
     "nickname": name
   }]
   result = callPrivateRPC("setContactLocalNickname".prefix, payload)
 
-proc addContact*(id: string, ensName: string): RpcResponse[JsonNode] {.raises: [Exception].} =
+proc sendContactRequest*(id: string, message: string): RpcResponse[JsonNode] {.raises: [Exception].} =
   let payload = %* [{
       "id": id,
-      "ensName": ensName
+      "message": message
     }]
-  result = callPrivateRPC("addContact".prefix, payload)
+  result = callPrivateRPC("sendContactRequest".prefix, payload)
+
+proc acceptContactRequest*(id: string): RpcResponse[JsonNode] {.raises: [Exception].} =
```

Comment:
@saledjenic Here is an another problem. `acceptContactRequest` need to pass the id of the contact request, not of contact. But we don't have entity for contact request. What sould i do, add it to contact as optional field?

Reply:
@MishkaRogachev feel free to add new hash map which will map appropriate contact id to appropriate contact request id, that way you will know when you accept somebody's contact request what contact request id is appropriate one for his contact id. But also pay attention if somebody doesn't accept contact request once he receives it, to populate that map correctly on the next app run. That means you need to check do we have a new rpc endpoint which will return currently active contact requests ids for logged in users or maybe that is added as part of fetch all contacts call we're doing on the app start.

Reply:
@michaelsbradleyjr here i ned to store map contact.id-contact_request.id. How can I promoute contact request signal from activity center to conatcts/service.nim?

Reply:
@MishkaRogachev why do you need to get anything from activity center? AC doesn't need to determine anything in the app, it stores only notifications which are generated by logic beyond AC. So if you need the list of contact requests then you should check do we receive that list as a part of fetch all contacts or is there any special endpoint where you can get the list of pending contact requests.

Reply:
@MishkaRogachev , AC is a separate thing. It also should receive information about contact requests, but only to show them in AC list. I see that contact_service is another endpoint which receives contact requests and probably this is a part you should focus on.